packages/react-components/react-color-picker-preview/library/src/components/ColorArea/ColorArea.types.ts

```diff
@@ -1,17 +1,52 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type { ComponentState, Slot, EventHandler, EventData, ComponentProps } from '@fluentui/react-utilities';
+
+export type ColorAreaOnColorChangeData = EventData<'change', React.SyntheticEvent | MouseEvent> & {
+  color: HsvColor;
+};
 
 export type ColorAreaSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  thumb?: NonNullable<Slot<'div'>>;
+  inputX?: NonNullable<Slot<'input'>>;
+  inputY?: NonNullable<Slot<'input'>>;
+};
+
+type ColorAreaInternalSlots = {
```

Comment:
Why do we need both `ColorAreaSlots` and `ColorAreaInternalSlots`? Is it possible to combine them using `Partial` for props and `Required` for the state?